src/main/java/gregtech/api/pipenet/block/BlockPipe.java

```diff
@@ -325,23 +326,28 @@ public boolean onPipeActivated(World world, IBlockState state, BlockPos pos, Ent
             }
         }
 
-        EnumFacing coverSide = ICoverable.traceCoverSide(hit);
-        if (coverSide == null) {
+        EnumFacing coverSide = ICoverable.determineGridSideHit(hit);
+        CoverBehavior coverBehavior = pipeTile.getCoverableImplementation().getCoverAtSide(coverSide);
+        
+        if (coverBehavior == null) {
             return activateFrame(world, state, pos, entityPlayer, hand, hit, pipeTile);
         }
 
         if (!(hit.cuboid6.data instanceof CoverSideData)) {
-            switch (onPipeToolUsed(world, pos, itemStack, coverSide, pipeTile, entityPlayer)) {
+            switch (onPipeToolUsed(world, pos, itemStack, coverSide, pipeTile, entityPlayer, hand)) {
                 case SUCCESS:
                     return true;
                 case FAIL:
                     return false;
             }
         }
 
-        CoverBehavior coverBehavior = pipeTile.getCoverableImplementation().getCoverAtSide(coverSide);
-        if (coverBehavior == null) {
-            return activateFrame(world, state, pos, entityPlayer, hand, hit, pipeTile);
```

Comment:
Same as above, this check needs to remain here for the pipe/cable connections. The soft mallet check can be inserted right after this.

Reply:
should be okay now